src/wp-includes/class-wp-theme-json.php

```diff
@@ -2818,6 +2823,14 @@ static function ( $pseudo_selector ) use ( $selector ) {
 		// 6. Generate and append the style variation rulesets.
 		foreach ( $style_variation_declarations as $style_variation_selector => $individual_style_variation_declarations ) {
 			$block_rules .= static::to_ruleset( ":root :where($style_variation_selector)", $individual_style_variation_declarations );
+			if ( isset( $style_variation_custom_css[ $style_variation_selector ] ) ) {
+				$block_rules .= $style_variation_custom_css[ $style_variation_selector ];
+			}
+		}
+
+		// 7. Generate and append any custom CSS rules pertaining to nested block style variations.
+		if ( isset( $node['css'] ) && ! $is_root_selector ) {
```

Comment:
Because in core we're not setting the `css` property in `get_block_nodes`, this will only be true for nested block style variations.